docker/Dockerfile

```diff
@@ -251,21 +292,37 @@ RUN --mount=type=bind,from=conda_bld_morpheus,source=/opt/conda/conda-bld,target
     /opt/conda/bin/conda env update --solver=libmamba -n morpheus --file \
         conda/environments/runtime_cuda-${CUDA_MAJOR_VER}${CUDA_MINOR_VER}_arch-x86_64.yaml
 
+# ============ Stage: git_clone ============
+# Perform a clone of the git repository this ensures that when we copy files from the source repository, we aren't
+# unintentionally including build artifacts or other files that shouldn't be in the final image
+FROM conda_env_dev as git_clone
+
+ARG GIT_URL
+ARG GIT_BRANCH
+ARG GIT_COMMIT
+
+RUN source activate morpheus &&\
+    git clone --branch ${GIT_BRANCH} ${GIT_URL} /tmp/morpheus_repo &&\
+    cd /tmp/morpheus_repo &&\
+    git checkout ${GIT_COMMIT} &&\
+    git lfs install &&\
+    /tmp/morpheus_repo/scripts/fetch_data.py fetch examples models
```

Comment:
This seems like it could be improved by doing something similar to what conda does and clone from a local git repo. This would avoid needing to specify the URL, branch and commit. It would just use whatever is currently cloned.

Should be something like `git clone file:///path/to/repo.git/`